src/System.Runtime/tests/System/Char.cs

```diff
@@ -5,516 +5,502 @@
+        VerifyConvertToUtf32("\u0020\uD7FF", 1, 0xD7FF);
+        VerifyConvertToUtf32("abcde", 4, 'e');
+
+        VerifyConvertToUtf32("\uD800\uD7FF", 1, 0xD7FF);  // High, non-surrogate
+        VerifyConvertToUtf32("\uD800\u0000", 1, 0);  // High, non-surrogate
+        VerifyConvertToUtf32("\uDF01\u0000", 1, 0);  // Low, non-surrogate
     }
 
-    [Fact]
-    public static void TestConvertToUtf32()
+    private static void VerifyConvertToUtf32(string s, int index, int expected)
     {
-        // Int32 Char.ConvertToUtf32(Char, Char)
-        ValidateconverToUtf32<Exception>('\uD800', '\uDC00', 0x10000);
-        ValidateconverToUtf32<Exception>('\uD800', '\uDFFF', 0x103FF);
-        ValidateconverToUtf32<Exception>('\uDBBF', '\uDFFF', 0xFFFFF);
-        ValidateconverToUtf32<Exception>('\uDBFF', '\uDC00', 0x10FC00);
-        ValidateconverToUtf32<Exception>('\uDBFF', '\uDFFF', 0x10FFFF);
+        Assert.Equal(expected, char.ConvertToUtf32(s, index));
+    }
 
-        ValidateconverToUtf32<ArgumentOutOfRangeException>('\uD800', '\uD800', int.MinValue);  // high, high
-        ValidateconverToUtf32<ArgumentOutOfRangeException>('\uD800', '\uD7FF', int.MinValue);  // high, non-surrogate
-        ValidateconverToUtf32<ArgumentOutOfRangeException>('\uD800', '\u0000', int.MinValue);  // high, non-surrogate
-        ValidateconverToUtf32<ArgumentOutOfRangeException>('\uDC01', '\uD940', int.MinValue);  // low, high
-        ValidateconverToUtf32<ArgumentOutOfRangeException>('\uDD00', '\uDE00', int.MinValue);  // low, low
-        ValidateconverToUtf32<ArgumentOutOfRangeException>('\uDF01', '\u0000', int.MinValue);  // low, non-surrogate
-        ValidateconverToUtf32<ArgumentOutOfRangeException>('\u0032', '\uD7FF', int.MinValue);  // both non-surrogate
-        ValidateconverToUtf32<ArgumentOutOfRangeException>('\u0000', '\u0000', int.MinValue);  // both non-surrogate
+    [Fact]
+    public static void TestConvertToUtf32_String_Int_Invalid()
+    {
+        Assert.Throws<ArgumentNullException>("s", () => char.ConvertToUtf32(null, 0)); // String is null
+
+        Assert.Throws<ArgumentException>("s", () => char.ConvertToUtf32("\uD800\uD800", 0)); // High, high
+        Assert.Throws<ArgumentException>("s", () => char.ConvertToUtf32("\uD800\uD7FF", 0)); // High, non-surrogate
+        Assert.Throws<ArgumentException>("s", () => char.ConvertToUtf32("\uD800\u0000", 0)); // High, non-surrogate
+        Assert.Throws<ArgumentException>("s", () => char.ConvertToUtf32("\uD800\uD800", 1)); // High, high
+
+        Assert.Throws<ArgumentException>("s", () => char.ConvertToUtf32("\uDC01\uD940", 0)); // Low, high
+        Assert.Throws<ArgumentException>("s", () => char.ConvertToUtf32("\uDD00\uDE00", 0)); // Low, low
+        Assert.Throws<ArgumentException>("s", () => char.ConvertToUtf32("\uDF01\u0000", 0)); // Low, non-surrogate
+        Assert.Throws<ArgumentException>("s", () => char.ConvertToUtf32("\uDC01\uD940", 1)); // Low, high
```

Comment:
Nit: in the "\uD800\u0000" case, you reorganized the lines so that both cases of it were next to each other, but here you didn't.  It'd be good to be consistent.